src/color.js

```diff
@@ -267,10 +267,10 @@ var CPK_TABLE = {
  FE : [0.56, 0.31, 0.12]
 };
 
-exports.byElement = function() {
+exports.byElement = function(palette) {
   return new ColorOp(function(atom, out, index) {
     var ele = atom.element();
-    var color = CPK_TABLE[ele];
+    var color = palette ? palette[ele] : CPK_TABLE[ele];
```

Comment:
Since the palette is either provided or omitted, you could move the palette selection out of this function directly into exports.byElement. This avoids doing the redundant palette selection for every atom.